### PR TITLE
Fix several index entries.

### PR DESCRIPTION
**Pull request by Eelis**

### CODE REVIEW

File: source/lex.tex [hunk 1]
```diff
@@ -28,7 +28,7 @@
 \indextext{conventions!lexical|(}%
 \indextext{compilation!separate|(}%
 The text of the program is kept in units called
-\indextext{source file}\term{source files} in this International
+\defnx{source files}{source file} in this International
```
Comment (zygoloid) [inline source/lex.tex]: \defn{source file}{s} perhaps?

**Comment by zygoloid** [comment]
The diffs look good to me, thanks.

@tkoeppe Would you mind looking over the built standard to double-check?

File: source/grammar.tex [hunk 2]
```diff
@@ -3,8 +3,8 @@
 \begin{paras}
 
 \pnum
-\index{grammar}%
-\index{summary, syntax}%
+\indextext{grammar}%
+\indextext{summary, syntax}%
```
Comment (tkoeppe) [inline source/grammar.tex]: Is there a reason this is `summary, syntax` rather than `summary|syntax`?

File: source/strings.tex [hunk 3]
```diff
@@ -6,9 +6,9 @@
 \pnum
 This Clause describes components for manipulating sequences of
 any non-array POD~(\ref{basic.types}) type.
-In this Clause such types are called \term{char-like types},\indextext{char-like type}
+In this Clause such types are called \defn{char-like types}{char-like type},
```
Comment (tkoeppe) [inline source/strings.tex]: Are you sure `\defn` works as intended here? Shouldn't it also be `\defnx`?

**Comment by tkoeppe** [comment]
Generally a good change. Two nitpicks, see above. The change replaces the low-level `\indextext` followed by a `\term` with the combined `\defnx` that combines indexing and rendering into a single command. It also fixes a few errors.

**Comment by Eelis** [comment]
Thanks for the review and suggestions! I've re-pushed the branch with amended commits fixing the issues.

**Comment by tkoeppe** [comment]
The changes look good!

**Comment by Eelis** [comment]
Ah, thanks, I'll look into it.

**Comment by Eelis** [comment]
It was simply because [except.ctor]p1 actually defines "stack unwinding" as a term with a proper index entry. So there is no need for a "see" index entry in the first place. :)

I've amended the commit that that tried to fix that "see" index entry, to no longer try to fix it, and added a separate commit to remove it.

**Comment by tkoeppe** [comment]
Sounds good, thanks! @zygoloid: I think this is ready for merging.

File: source/intro.tex [hunk 4]
```diff
@@ -2,7 +2,7 @@
 \rSec0[intro]{General}
 
 \indextext{diagnostic message|see{message, diagnostic}}%
-\indexdefn{conditionally-supported behavior!see{behavior, conditionally-supported}}%
+\indexdefn{conditionally-supported behavior|see{behavior, conditionally-supported}}%
```
Comment (burblebee) [inline source/intro.tex]: When to use indexdefn vs. indextext?  Looking in macros.tex, they seem to be equivalent.

**Comment by jwakely** [comment]
> When to use indexdefn vs. indextext? Looking in macros.tex, they seem to be equivalent.

Currently, yes, but indexdefn has more semantic meaning, and could in theory be formatted differently in future.

**Pull request merged**